include/tvm/tir/index_map.h

```diff
@@ -133,16 +145,19 @@ class IndexMapNode : public Object {
   void VisitAttrs(AttrVisitor* v) {
     v->Visit("initial_indices", &initial_indices);
     v->Visit("final_indices", &final_indices);
+    v->Visit("inverse_index_map", &inverse_index_map);
   }
 
   bool SEqualReduce(const IndexMapNode* other, SEqualReducer equal) const {
     return equal.DefEqual(initial_indices, other->initial_indices) &&
-           equal(final_indices, other->final_indices);
+           equal(final_indices, other->final_indices) &&
+           equal(inverse_index_map, other->inverse_index_map);
```

Comment:
Do we need to include `inverse_index_map` in structural equality check? Consider a case where this field is optional so `lhs` doesn't have it while `rhs` does, and should they be considered the same?